src/realm/collection_parent.hpp

```diff
@@ -71,40 +71,44 @@ enum class UpdateStatus {
     /// current state.
     NoChange,
 };
-
 struct PathElement {
```

Comment:
Maybe we can use a comment here to explain why we need  `PathElement` for a nested collection or a collection in mixed.